src/proxy/Map.ts

```diff
@@ -56,4 +124,269 @@ export class Map<K, V> extends BaseProxy implements IMap<K, V> {
     isEmpty(): Q.Promise<boolean> {
         return this.encodeInvokeOnRandomTarget<boolean>(MapIsEmptyCodec);
     }
+
+    getAll(keys: K[]): Q.Promise<any[]> {
+        var partitionService = this.client.getPartitionService();
+        var partitionsToKeys: {[id: string]: any} = {};
+        var key: K;
+        for (var i in keys) {
+            key = keys[i];
```

Comment:
Since we use Typescript compiler target as `ES5` all `let`s are transpiled to `var`. So this would be worse. You expect it to be block scoped and get function scoped. Since Node 4 do not support majority of ES6 features we have to stick with ES5 and `var`s.